### PR TITLE
Remove tmp sync rando et sync mobile

### PR DESCRIPTION
**Pull request by LePetitTim**

### CODE REVIEW

**Comment by LePetitTim** [comment]
On enleve du code : la couverture dimminue

**Review by Chatewgne — APPROVED** [review]

File: geotrek/api/management/commands/sync_mobile.py [hunk 1]
```diff
@@ -87,7 +88,7 @@ def sync_view(self, lang, view, name, url='/', params=None, headers={}, zipfile=
         # If new file is identical to old one, don't recreate it. This will help backup
         if os.path.isfile(oldfilename) and filecmp.cmp(fullname, oldfilename):
             os.unlink(fullname)
-            os.link(oldfilename, fullname)
+            shutil.copy(oldfilename, fullname)
```
Comment (submarcos) [inline geotrek/api/management/commands/sync_mobile.py:91]: as comment says, os.link is used to don't recreate file and use and [hardlink](https://docs.python.org/3/library/os.html#os.link) 
This save many drive space on sync

File: geotrek/api/management/commands/sync_mobile.py [hunk 2]
```diff
@@ -160,7 +161,7 @@ def sync_file(self, name, src_root, url, directory='', zipfile=None):
         dst = os.path.join(self.tmp_root, directory, url, name)
         self.mkdirs(dst)
         if not os.path.isfile(dst):
-            os.link(src, dst)
+            shutil.copy(src, dst)
```
Comment (submarcos) [inline geotrek/api/management/commands/sync_mobile.py:164]: idem

**Review by LePetitTim — COMMENTED** [review]
Purge sync_rando tmp/sync_rando dans le dpkg-reconfigure / update.sh 
+ option purge tmp/sync_rando

File: geotrek/api/management/commands/sync_mobile.py [hunk 3]
```diff
@@ -36,6 +37,8 @@
 class Command(BaseCommand):
     def add_arguments(self, parser):
         parser.add_argument('path')
+        parser.add_argument('--empty-tmp-folder', dest='empty_tmp_folder', action='store_true', default=False,
+                            help='Empty tmp folders')
```
Comment (submarcos) [inline geotrek/api/management/commands/sync_mobile.py:41]:
```suggestion
                            help='Empty tmp folder')
```

File: geotrek/api/tests/test_mobile/test_views.py [hunk 4]
```diff
@@ -70,30 +69,26 @@ def test_get_sync_mobile_states_superuser_with_sync_mobile(self, mocked_stdout):
     @patch('sys.stdout', new_callable=StringIO)
     @patch('geotrek.api.management.commands.sync_mobile.Command.handle', return_value=None,
            side_effect=Exception('This is a test'))
-    @override_settings(SYNC_MOBILE_ROOT='tmp', SYNC_MOBILE_OPTIONS={'url': 'http://localhost:8000',
-                                                                    'skip_tiles': True})
+    @override_settings(SYNC_MOBILE_ROOT=os.path.join(settings.VAR_DIR, 'tmp', 'tmp_sync'),
```
Comment (submarcos) [inline geotrek/api/tests/test_mobile/test_views.py:72]:
```suggestion
    @override_settings(SYNC_MOBILE_ROOT=os.path.join(settings.TMP_DIR, 'tmp_sync'),
```

File: geotrek/api/tests/test_mobile/test_views.py [hunk 5]
```diff
@@ -105,7 +100,7 @@ def test_launch_sync_mobile_fail(self, mocked_stdout, command):
         self.assertNotIn('Sync mobile ended', log)
         self.assertEqual(task.status, "FAILURE")
 
-    @override_settings(SYNC_MOBILE_ROOT='tmp')
+    @override_settings(SYNC_MOBILE_ROOT=os.path.join(settings.VAR_DIR, 'tmp', 'tmp_sync'))
```
Comment (submarcos) [inline geotrek/api/tests/test_mobile/test_views.py:103]:
```suggestion
    @override_settings(SYNC_MOBILE_ROOT=os.path.join(settings.TMP_DIR, 'tmp_sync'))
```

File: geotrek/common/management/commands/sync_rando.py [hunk 6]
```diff
@@ -502,16 +505,13 @@ def handle(self, *args, **options):
...
-        try:
-            os.mkdir(self.tmp_root)
-        except OSError as e:
-            if e.errno != 17:
-                raise
-            raise CommandError(
-                "The {}/ directory already exists. Please check no other sync_rando command is already running."
-                " If not, please delete this directory.".format(self.tmp_root)
-            )
+        if options['empty_tmp_folder']:
```
Comment (submarcos) [inline geotrek/common/management/commands/sync_rando.py:508]:
```suggestion
        sync_rando_tmp_dir = os.path.join(settings.TMP_DIR, 'sync_rando')
        if options['empty_tmp_folder']:
```
os.path.join only once

File: geotrek/common/tasks.py [hunk 7]
```diff
@@ -132,6 +132,10 @@ def launch_sync_rando(*args, **kwargs):
     """
     celery shared task - sync rando command
     """
+    if not os.path.exists(settings.TMP_DIR):
```
Comment (submarcos) [inline geotrek/common/tasks.py:135]: use attrivbute and os.path.join only once

**Pull request merged**